activerecord/test/cases/fixtures_test.rb

```diff
@@ -833,29 +833,42 @@ class TransactionalFixturesOnConnectionNotification < ActiveRecord::TestCase
   self.use_instantiated_fixtures = false
 
   def test_transaction_created_on_connection_notification
-    connection = stub(transaction_open?: false)
+    connection = Class.new do
+      attr_accessor :pool
+
+      def transaction_open?; end
+      def begin_transaction(*args); end
+      def rollback_transaction(*args); end
+    end.new
+
+    connection.pool = Class.new do
+      def lock_thread=(lock_thread); false; end
```

Comment:
returning `false` seems extra to me. There is one more occurrence below.  See diff:
```diff
diff --git a/activerecord/test/cases/fixtures_test.rb b/activerecord/test/cases/fixtures_test.rb
index 2d8cbed24e..2ead2c4816 100644
--- a/activerecord/test/cases/fixtures_test.rb
+++ b/activerecord/test/cases/fixtures_test.rb
@@ -842,7 +842,7 @@ def rollback_transaction(*args); end
     end.new

     connection.pool = Class.new do
-      def lock_thread=(lock_thread); false; end
+      def lock_thread=(lock_thread); end
     end.new

     connection.expects(:begin_transaction).with(joinable: false)
@@ -863,7 +863,7 @@ def rollback_transaction(*args)
     end.new

     connection.pool = Class.new do
-      def lock_thread=(lock_thread); false; end
+      def lock_thread=(lock_thread); end
     end.new

     fire_connection_notification(connection)

```